### PR TITLE
[W000] Bootstrap federated semantic trace scaffolding

### PR DESCRIPTION
**Pull request by GBOGEB**
### Motivation
- Introduce a W000 bootstrap to bridge ABACUS ↔ CODEX and provide a first-stage dual-render governance model prioritizing human semantic readability while preserving machine-sequential telemetry.
- Provide a minimal, discoverable semantic prefix and tuple schema to enable grep-friendly, LLM-retrievable session tuples and completion-vector scoring.
- Establish a telemetry foothold for drift detection between semantic and temporal layers so future waves can extend PCA-based monitoring and gating.

### Description
- Added a devcontainer scaffold at `/.devcontainer/devcontainer.json` to standardize onboarding and tooling for the wave; updated `postCreateCommand` to fail fast on dependency errors and mirror CI by running `pip install -e .`; merged additional settings from main branch including `files.insertFinalNewline` and `redhat.vscode-yaml` extension.
- Added a semantic index schema at `federation/semantic_index/schema.yaml` defining `semantic_prefixes`, `session_tuple` key format, required fields, and `completion_vector` ranges.
- Added a PCA drift monitor bootstrap at `telemetry/pca/drift_monitor.py` implementing `compute_drift_score` and `evaluate_drift` and returning a `DriftSignal` dataclass; added `telemetry/__init__.py` and `telemetry/pca/__init__.py` to make the package explicit, consistent with the rest of the repo.
- Added comprehensive unit tests at `tests/test_drift_monitor.py` (14 tests) covering edge cases, empty vectors, mismatched lengths, threshold boundaries, single elements, and iterables; updated immutability assertion to use `dataclasses.FrozenInstanceError` and switched all float comparisons to `pytest.approx` for consistency.
- Added agent protocol docs `agents/codex/MCP_INSTRUCTION.md` and `agents/abacus/FEDERATION_PROTOCOL.md`, a runtime incubator placeholder `runtime/incubator/.gitkeep`, and updated the PR template at `.github/pull_request_template.md` with placeholder values (`[WAVE_ID]`, `[TRACE_NAME]`, `[0-100%]`, `[NEARMISS_N] ...`, etc.).
- Updated `README.md` to integrate W000 bootstrap section with GitHub interface package description, documenting all bootstrap artifacts including `runtime/incubator/.gitkeep`.
- Resolved merge conflicts with main branch, preserving all review feedback fixes while integrating upstream changes from main.

### Testing
- Executed the drift monitor with `python telemetry/pca/drift_monitor.py`, which produced a numeric `score` and `stable: True`, indicating the bootstrap drift evaluation ran successfully.
- All 14 unit tests pass for the drift monitor (`pytest tests/test_drift_monitor.py`).
- Manifest check passes (`python scripts/check_manifest.py`).
- Code Review validation passes with no issues.
- CodeQL Security Scan passes with 0 alerts.
- Merge with main branch completed successfully with all tests passing.

------
<a href="https://chatgpt.com/codex/cloud/tasks/task_b_6a15b5c16568832ebcb993f553a6109c">Codex Task</a>

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "python.defaultInterpreterPath": "/usr/local/bin/python"
+      },
+      "extensions": [
+        "ms-python.python",
+        "ms-toolsai.jupyter",
+        "yzhang.markdown-all-in-one"
+      ]
+    }
+  },
+  "postCreateCommand": "pip install -r requirements.txt || true"
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply .devcontainer/devcontainer.json]: @copilot fix and commit

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
-## Lineage
-Program: <program>
-Repository: CODEX
+| Key | Value |
+|---|---|
+| Wave | W000 |
+| Trace | FEDERATION.RUNTIME |
+| State | ACTIVE |
+| Renderable | YES |
+| Drift Stable | YES |
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply .github/pull_request_template.md]: @copilot fix and commit

File: README.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+- Completion-vector oriented runtime scoring
+
+Bootstrap artifacts are staged in:
+- `.devcontainer/devcontainer.json`
+- `federation/semantic_index/schema.yaml`
+- `telemetry/pca/drift_monitor.py`
+- `agents/codex/MCP_INSTRUCTION.md`
+- `agents/abacus/FEDERATION_PROTOCOL.md`
+- `.github/pull_request_template.md`
+
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply README.md:22]: @copilot fix and commit

File: telemetry/pca/drift_monitor.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+def compute_drift_score(baseline: Iterable[float], observed: Iterable[float]) -> float:
+    """Return mean absolute deviation between baseline and observed vectors."""
+    baseline_list = list(baseline)
+    observed_list = list(observed)
+    if len(baseline_list) != len(observed_list):
+        raise ValueError("baseline and observed vectors must have equal length")
+    if not baseline_list:
+        raise ValueError("input vectors must not be empty")
+    deviation = [abs(a - b) for a, b in zip(baseline_list, observed_list)]
+    return sum(deviation) / len(deviation)
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply telemetry/pca/drift_monitor.py:30]: @copilot fix and commit

**Review by GBOGEB — COMMENTED** [review]
@copilot fix and merge all open, fix conflicts

**Review by GBOGEB — COMMENTED** [review]
Ok

**Review by GBOGEB — COMMENTED** [review]
Ok

File: telemetry/pca/drift_monitor.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+"""[WAVE.W000] PCA drift monitor bootstrap.
+
+Tracks drift between semantic and temporal layers using a minimal numerical
+proxy that can be extended in W002.
+"""
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply telemetry/pca/drift_monitor.py:5]: @copilot fix and commit

File: tests/test_drift_monitor.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import pytest
+
+from telemetry.pca.drift_monitor import DriftSignal, compute_drift_score, evaluate_drift
+
+
+def test_drift_signal_frozen() -> None:
+    """DriftSignal is frozen and immutable."""
+    signal = DriftSignal(score=0.1, stable=True)
+    with pytest.raises(Exception):
+        signal.score = 0.2  # type: ignore[misc]
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply tests/test_drift_monitor.py:14]: @copilot fix and commit

File: tests/test_drift_monitor.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    """Raises ValueError when baseline and observed have different lengths."""
+    with pytest.raises(ValueError, match="baseline and observed vectors must have equal length"):
+        compute_drift_score([0.9, 0.75], [0.88, 0.71, 0.5])
+
+
+def test_compute_drift_score_single_element() -> None:
+    """Drift score works correctly with single-element vectors."""
+    baseline = [0.9]
+    observed = [0.7]
+    assert abs(compute_drift_score(baseline, observed) - 0.2) < 1e-6
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply tests/test_drift_monitor.py]: @copilot fix and commit

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -1,16 +1,50 @@
...
+
+# Completion Vector
+
+| Domain | Completion |
+|---|---|
+| Structure | 90% |
+| Federation | 40% |
+| Telemetry | 35% |
+| Renderability | 75% |
+
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply .github/pull_request_template.md:33]: @copilot fix and commit

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -1,16 +1,50 @@
...
+---
+
+# Near-Miss Optimizations
+
+-
+-
+-
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply .github/pull_request_template.md]: @copilot fix and commit

File: tests/test_drift_monitor.py [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+
+def test_evaluate_drift_threshold_boundary() -> None:
+    """evaluate_drift correctly handles threshold boundary condition."""
+    baseline = [0.0, 1.0]
+    observed = [0.15, 1.15]
+    # Score = (0.15 + 0.15) / 2 = 0.15
+    signal = evaluate_drift(baseline, observed, threshold=0.15)
+    assert abs(signal.score - 0.15) < 1e-6
+    assert signal.stable is True  # stable when score <= threshold
```
[the remark this answers is not in the document]
Comment (GBOGEB) [reply tests/test_drift_monitor.py:78]: @copilot fix and commit

**Review by GBOGEB — COMMENTED** [review]
Ok
